tests/certification/configuration/postgres/postgres_test.go

```diff
@@ -0,0 +1,314 @@
+	"github.com/dapr/components-contrib/tests/certification/embedded"
+	"github.com/dapr/components-contrib/tests/certification/flow"
+	"github.com/dapr/components-contrib/tests/certification/flow/dockercompose"
+	"github.com/dapr/components-contrib/tests/certification/flow/network"
+	"github.com/dapr/components-contrib/tests/certification/flow/retry"
+	"github.com/dapr/components-contrib/tests/certification/flow/sidecar"
+	cu_postgres "github.com/dapr/components-contrib/tests/utils/configupdater/postgres"
+	configuration_loader "github.com/dapr/dapr/pkg/components/configuration"
+	"github.com/dapr/dapr/pkg/runtime"
+	dapr_testing "github.com/dapr/dapr/pkg/testing"
+	dapr "github.com/dapr/go-sdk/client"
+	"github.com/dapr/kit/logger"
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/components-contrib/metadata"
+	"github.com/dapr/components-contrib/tests/certification/flow/watcher"
+)
+
+const (
+	dockerComposeYAML   = "docker-compose.yml"
+	storeName           = "configstore"
+	key1                = "key1"
+	key2                = "key2"
+	Val1                = "val1"
+	Val2                = "val2"
+	sidecarName1        = "dapr-1"
+	sidecarName2        = "dapr-2"
+	connectionString    = "host=localhost user=postgres password=example port=5432 connect_timeout=10 database=dapr_test"
+	configTable         = "configTable"
+	connectionStringKey = "connectionString"
+	configTableKey      = "table"
+	configTable2        = "configTable2"
+	pgNotifyChannelKey  = "pgNotifyChannel"
+	pgNotifyChannel     = "config"
+	portOffset          = 2
+	channel1            = "channel1"
+	channel2            = "channel2"
+)
+
+var (
+	runID   = strings.ReplaceAll(uuid.Must(uuid.NewRandom()).String(), "-", "_")
+	counter = 0
+)
+
+// Cast go-sdk ConfigurationItem to contrib ConfigurationItem
+func castConfigurationItems(items map[string]*dapr.ConfigurationItem) map[string]*configuration.Item {
+	configItems := make(map[string]*configuration.Item)
+	for key, item := range items {
+		configItems[key] = &configuration.Item{
+			Value:    item.Value,
+			Version:  item.Version,
+			Metadata: item.Metadata,
+		}
+	}
+	return configItems
+}
+
+// Create UpdateEvent struct from given key, val pair
+func getUpdateEvent(key string, val string) configuration.UpdateEvent {
+	expectedUpdateEvent := configuration.UpdateEvent{
+		Items: map[string]*configuration.Item{
+			key: {
+				Value: val,
+			},
+		},
+	}
+	return expectedUpdateEvent
+}
+
+func TestPostgres(t *testing.T) {
+	log := logger.NewLogger("dapr.components")
+
+	// Updater client to update config table
+	updater := cu_postgres.NewPostgresConfigUpdater(log).(*cu_postgres.ConfigUpdater)
+
+	ports, err := dapr_testing.GetFreePorts(2)
+	assert.NoError(t, err)
+	currentGrpcPort := ports[0]
+	currentHTTPPort := ports[1]
+
+	watcher1 := watcher.NewUnordered()
```

Comment:
Configuration stores don't guarantee order of updates - so this is generally ok. Though since we are only dealing with updates to the same key where order does actually matter.. this probably needs to be updated to `Ordered()` for the sake of this test.

Reply:
Yes config stores don't guarantee order of updates. For a single key, there would be just one update notification and so ordered and unordered are same. 
I have updated the test to subscribe to multiple keys for more robust test.